src/Repository/IlluminateTenantRepository.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Repository;
+
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Model;
+use Stancl\Tenancy\Contracts\Tenant;
+
+class IlluminateTenantRepository implements TenantRepository
```

Comment:
What is this class meant to represent?

Also:
1. What's the point of the `TenantRepository` interface?
2. The naming here implies that this is a part of Laravel (`Illuminate`)

Reply:
Having a tenant (and domain) repository is very useful for testing. Because it allows you to test on one connection (`tenant`) because you can swap out the repository for the models in the central one for a fake implementation.

Reply:
Can you give a specific example?

Reply:
If you can use fake models for the central connection you'll only need one database which, among other things, allows you to do parallel testing.